game.py

```diff
@@ -0,0 +1,291 @@
+        return gameExit
+	
+    def add_icon(self,icon,location):
+        self.game_display.blit(icon,location)
+	#set starting spot to center
+    def get_center(self):
+        return (self.display_width/2, self.display_height/2)
+	#identify if unicorn intersects carrot.
+    def did_collide(self,obj1, obj2):
+        if obj1.x > obj2.x and obj1.x < obj2.x + obj2.image_width or obj1.x + obj1.image_width > obj2.x and obj1.x + obj1.image_width < obj2.x + obj2.image_width:
+            if obj1.y > obj2.y and obj1.y < obj2.y + obj2.image_height:
+                return True
+            elif obj1.y + obj1.image_height > obj2.y and obj1.y + obj1.image_height < obj2.y + obj2.image_height:
+                return True
+        return False
+
+#display_width,display_height, caption='Unicorn Squad', FPS=10
+
+class SnakeGame(GameManager):
+    def __init__(self,**kwargs):
+        super(SnakeGame,self).__init__(**kwargs)
+
+    def is_game_over(self,snake):
+        if snake.x >= self.display_width or snake.x < 0 or snake.y >= self.display_height or snake.y < 0:
+            return True
+        return False
+
+    def fill_blue(self):
+        self.game_display.fill(self.sky_blue)
+
+    # Initliazes text object
+    def text_objects(self, text, color):
+        textSurface = self.font.render(text, True, color)
+        return textSurface, textSurface.get_rect()
+
+    # Defines  display message with color variable
+    def message_to_screen(self, msg, color, y_displace=0):
+        textSurf, textRect = self.text_objects(msg,color)
+        textRect.center = (self.display_width / 2), (self.display_height / 2) + y_displace
+        self.game_display.blit(textSurf, textRect)
+    #places score on bottom right corner
+    def score_to_screen(self, msg, color):
+        textSurf, textRect = self.text_objects(msg, color)
+        textRect = (self.display_width - 50), (self.display_height - 50)
+        self.game_display.blit(textSurf, textRect)
+
+class Image(object):
+    def __init__(self,**kwargs):
+        self.start_pos = kwargs.pop('start_pos') #(x,y)
+        self.icon = pygame.image.load(kwargs.pop('icon'))
+        self.icon = pygame.transform.scale(self.icon, (kwargs.pop('image_width'), kwargs.pop('image_height')))
+        self.x = self.start_pos[0]
+        self.y = self.start_pos[1]
+        if 'image_width' in kwargs:  
+            self.image_width = kwargs.pop('image_width')
+        else:
+            self.image_width = 50 
+        if 'image_height' in kwargs:  
+            self.image_height = kwargs.pop('image_height')
+        else:
+            self.image_height = 50
+
+
+#start_pos=(0,0),icon='food.png', image_width=50, image_height=50
+
+#init "food" image
+class Carrot(Image):
+    def __init__(self,**kwargs):
+        super(Carrot,self).__init__(**kwargs)
+    
+    def set_random_location(self,display_width,display_height):
+        self.x = round(random.randrange(0, display_width-self.image_width))
```

Comment:
Many snake games avoid allowing targets to spawn "inside" the snake itself. Could be interesting to think about how one would go about doing that!